### PR TITLE
Trigger Multibranch Deployment

### PR DESCRIPTION
**Pull request by caleblloyd**
Prior to this, builds would only trigger a deployment on the `defaultbranch`

This PR adds `deploy.deploymentMap.${deployment}.triggerBranch` to allow triggers to go to different branches of a multibranch deployment

Also included some upgrade code to gradle to get this working with Java 11, and some code to fix tests that weren't working because `BaseConfig.equals` had a bug where it was always returning true

### CODE REVIEW

File: src/com/boxboat/jenkins/pipeline/BoxBase.groovy [hunk 1]
```diff
@@ -212,7 +220,12 @@ abstract class BoxBase<T extends CommonConfigBase> implements Serializable {
                 return
             }
             def configProperty = this.config.metaClass.getMetaProperty("$k")
-            if (configProperty && Modifier.isPublic(configProperty.modifiers) && !Modifier.isStatic(configProperty.modifiers) && this.config."$k" == null) {
+            def setMethodName = "set${k.capitalize()}"
```
Comment (boxboatmatt) [inline src/com/boxboat/jenkins/pipeline/BoxBase.groovy]: Doesn't look like you use this

File: src/com/boxboat/jenkins/pipeline/BoxBase.groovy [hunk 2]
```diff
@@ -142,7 +145,12 @@ abstract class BoxBase<T extends CommonConfigBase> implements Serializable {
                 return
             }
             def property = this.metaClass.getMetaProperty("$k")
-            if (property && Modifier.isPublic(property.modifiers) && !Modifier.isStatic(property.modifiers) && this."$k" == null) {
+            def setMethodName = "set${k.capitalize()}"
```
Comment (boxboatmatt) [inline src/com/boxboat/jenkins/pipeline/BoxBase.groovy]: Doesn't look like you use this

File: src/com/boxboat/jenkins/pipeline/BoxBase.groovy [hunk 3]
```diff
@@ -247,12 +260,13 @@ abstract class BoxBase<T extends CommonConfigBase> implements Serializable {
     }
 
     def writeTriggers() {
-        if (gitRepo.branch == config.defaultBranch && gitRepo.isBranchTip()) {
+        def triggers = this.triggers()
+        if (triggers != null && gitRepo.isBranchTip()) {
```
Comment (boxboatmatt) [inline src/com/boxboat/jenkins/pipeline/BoxBase.groovy]: Will this if block ever execute? `triggers()` always returns `null`

Comment (caleblloyd) [reply src/com/boxboat/jenkins/pipeline/BoxBase.groovy]: It's overridden in `BoxDeploy`

File: src/com/boxboat/jenkins/pipeline/BoxBase.groovy (hunk 1 above)
Comment (caleblloyd) [reply src/com/boxboat/jenkins/pipeline/BoxBase.groovy]: removed

File: src/com/boxboat/jenkins/pipeline/BoxBase.groovy (hunk 2 above)
Comment (caleblloyd) [reply src/com/boxboat/jenkins/pipeline/BoxBase.groovy]: :+1: removed

**Pull request merged**